Mathlib/Data/Matrix/Basis.lean

```diff
@@ -68,16 +68,14 @@ theorem mulVec_stdBasisMatrix [NonUnitalNonAssocSemiring α] [Fintype m]
   · simp
   simp [h, h.symm]
 
+-- In the following proof, we abuse the definition of `Matrix` by applying matrices.
```

Comment:
Is that abuse? Isn't that how matrices are designed to be used?

Reply:
It is abuse to the extent that `Matrix m n α` is not synteq to `m → n → α`. So simp doesn't fire lemmas about `m → n → α` on a subexpression of type `Matrix m n α`. This is precisely what's happening below.

In fact, I am surprised to see that `Matrix` doesn't have a `FunLike` instance. That would seem li